lib/openpgp_crypt_rsa.php

```diff
@@ -228,43 +230,73 @@ static function try_decrypt_session($key, $edata) {
   }
 
   static function crypt_rsa_key($mod, $exp, $hash='SHA256') {
-    $rsa = new Crypt_RSA();
-    $rsa->setSignatureMode(CRYPT_RSA_SIGNATURE_PKCS1);
-    $rsa->setHash(strtolower($hash));
-    $rsa->modulus = new Math_BigInteger($mod, 256);
-    $rsa->k = strlen($rsa->modulus->toBytes());
-    $rsa->exponent = new Math_BigInteger($exp, 256);
-    $rsa->setPublicKey();
-    return $rsa;
+
+    return Crypt_RSA::loadPublicKey([
+        'e' => new Math_BigInteger($exp, 256),
+        'n' => new Math_BigInteger($mod, 256),
+    ])
+        ->withPadding(CRYPT_RSA_SIGNATURE_PKCS1)
+        ->withHash(strtolower($hash));
+    ;
+
   }
 
   static function convert_key($packet, $private=false) {
     if(!is_object($packet)) $packet = OpenPGP_Message::parse($packet);
     if($packet instanceof OpenPGP_Message) $packet = $packet[0];
 
-    $mod = $packet->key['n'];
+
+
     $exp = $packet->key['e'];
     if($private) $exp = $packet->key['d'];
     if(!$exp) return NULL; // Packet doesn't have needed data
 
-    $rsa = self::crypt_rsa_key($mod, $exp);
+    /**
+     * @see https://github.com/phpseclib/phpseclib/issues/1113
+     * Primes and coefficients now use BigIntegers.
+     **/
 
     if($private) {
         /**
-         * @see https://github.com/phpseclib/phpseclib/issues/1113
-         * Primes and coefficients now use BigIntegers.
-         **/
-        //set the primes
-        if($packet->key['p'] && $packet->key['q'])
-            $rsa->primes = array(
-                1 => new Math_BigInteger($packet->key['p'], 256),
-                2 => new Math_BigInteger($packet->key['q'], 256)
-            );
-        // set the coefficients
-        if($packet->key['u']) $rsa->coefficients = array(2 => new Math_BigInteger($packet->key['u'], 256));
+         * @see https://github.com/phpseclib/phpseclib/issues/1711
+         */
+        $xml = sprintf(
```

Comment:
i know it's hackish , but I have found (see the phpseclib issue i've opened) a  more direct way to do it

Reply:
and actually I was wondering if this way of loading RSA private key was only because we use it like that in the examples  or there's a real case where we would have the RSA  *private key* as a raw array of parameters ?

Reply:
ok the issue has been fixed in phpseclib